### PR TITLE
[nrf52840] Improved nrf security - fix for first packet coming unencrypted

### PR DESCRIPTION
**Pull request by MarekPorwisz**
addresses https://github.com/openthread/openthread/issues/5767

Had to switch to latest nrf_security as the old one does not support ROM AES tables. But since the new one adds support for oberon backend, I made use of it as it is a lot faster than mbedtls.

### CODE REVIEW

**Comment by MarekPorwisz** [comment]
@piotrkoziar please also have a look at it. I might have missed something that I was not aware of.

**Review by LuDuda — APPROVED** [review]
LGTM :+1:

**Comment by jwhui** [comment]
@MarekPorwisz , the arm-gcc-4 build test is failing, can you take a look?

**Comment by MarekPorwisz** [comment]
Squashed all commits, as the prior commits would have some issues working without the last one (in a specific conditions only). All tests passed before the squash.

**Comment by jwhui** [comment]
@piotrkoziar , please review when you have a chance. Thanks!

File: third_party/NordicSemiconductor/libraries/nrf_security/config/nrf-config.h [hunk 1]
```diff
@@ -653,15 +743,15 @@
  *
  * Comment macros to disable the curve and functions for it
  */
-#define MBEDTLS_ECP_DP_SECP192R1_ENABLED
-#define MBEDTLS_ECP_DP_SECP224R1_ENABLED
+/* #undef MBEDTLS_ECP_DP_SECP192R1_ENABLED */
```
Comment (piotrkoziar) [inline third_party/NordicSemiconductor/libraries/nrf_security/config/nrf-config.h:746]: Please take a look at https://github.com/openthread/openthread/pull/5265. Do you know if the additional curves are present in DTLS client hello? If still, we should apply changes as in https://github.com/openthread/openthread/pull/5265.

**Review by piotrkoziar — APPROVED** [review]
LGTM

File: third_party/NordicSemiconductor/libraries/nrf_security/config/nrf-config.h (hunk 1 above)
Comment (MarekPorwisz) [reply third_party/NordicSemiconductor/libraries/nrf_security/config/nrf-config.h:746]: Discussed with @piotrkoziar and this is something that can be addressed at a later time. If this happens to be still applicable it is only a small inconsistency in the libs provided and configuration file.

**Review by jwhui — APPROVED** [review]
Thanks! 👍

**Pull request merged**